### PR TITLE
Bokeh 2.0 updates

### PR DESCRIPTION
**Pull request by philippjfr**
@mattpap Ignore my earlier ping, it all works now.

### CODE REVIEW

File: panel/io/notebook.py [hunk 1]
```diff
@@ -23,13 +23,18 @@
...
 from pyviz_comms import (
     JS_CALLBACK, PYVIZ_PROXY, Comm, JupyterCommManager as _JupyterCommManager,
     nb_mime_js
 )
 
+try:
+    from bokeh.util.string import encode_utf8, escape
+except:
+    from html import escape
+    encode_utf8 = lambda s: s
```
Comment (philippjfr) [inline panel/io/notebook.py]: Should really just remove this, we won't support py2 anyway.

**Comment by xavArtley** [comment]
Hello,

I tried to update bokeh to master and test this branch however when I built panel I encounter this error:
```
(py37) xavier:~/PythonWorkspace/modules/panel$ bokeh build panel --rebuild
Traceback (most recent call last):
  File "/home/xavier/PythonWorkspace/modules/bokeh/bokeh/util/dependencies.py", line 79, in import_required
    return import_module(mod_name)
  File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 965, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'selenium'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/xavier/PythonWorkspace/py37/bin/bokeh", line 11, in <module>
    load_entry_point('bokeh', 'console_scripts', 'bokeh')()
  File "/home/xavier/PythonWorkspace/modules/bokeh/bokeh/__main__.py", line 53, in main
    from bokeh.command.bootstrap import main as _main
  File "/home/xavier/PythonWorkspace/modules/bokeh/bokeh/command/bootstrap.py", line 54, in <module>
    from . import subcommands
  File "/home/xavier/PythonWorkspace/modules/bokeh/bokeh/command/subcommands/__init__.py", line 76, in <module>
    all = _collect()
  File "/home/xavier/PythonWorkspace/modules/bokeh/bokeh/command/subcommands/__init__.py", line 60, in _collect
    mod = import_module("." + modname, __package__)
  File "/usr/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/xavier/PythonWorkspace/modules/bokeh/bokeh/command/subcommands/html.py", line 78, in <module>
    from .file_output import FileOutputSubcommand
  File "/home/xavier/PythonWorkspace/modules/bokeh/bokeh/command/subcommands/file_output.py", line 32, in <module>
    from ..util import build_single_handler_applications, die
  File "/home/xavier/PythonWorkspace/modules/bokeh/bokeh/command/util.py", line 30, in <module>
    from bokeh.application.handlers import (
  File "/home/xavier/PythonWorkspace/modules/bokeh/bokeh/application/handlers/__init__.py", line 23, in <module>
    from .code import CodeHandler
  File "/home/xavier/PythonWorkspace/modules/bokeh/bokeh/application/handlers/code.py", line 44, in <module>
    from ...io.doc import curdoc, set_curdoc
  File "/home/xavier/PythonWorkspace/modules/bokeh/bokeh/io/__init__.py", line 23, in <module>
    from .export import export_png, export_svgs
  File "/home/xavier/PythonWorkspace/modules/bokeh/bokeh/io/export.py", line 38, in <module>
    from .webdriver import WebDriver, webdriver_control
  File "/home/xavier/PythonWorkspace/modules/bokeh/bokeh/io/webdriver.py", line 19, in <module>
    "To use bokeh.io image export functions you need selenium "
  File "/home/xavier/PythonWorkspace/modules/bokeh/bokeh/util/dependencies.py", line 81, in import_required
    raise RuntimeError(error_msg) from e
RuntimeError: To use bokeh.io image export functions you need selenium ('conda install selenium' or 'pip install selenium')
```
After installing selenium everything is ok however is it really a hard dependency?

**Comment by mattpap** [comment]
> After installing selenium everything is ok however is it really a hard dependency?

Start an issue in bokeh. This is a recent regression.

**Comment by philippjfr** [comment]
Reminder: Check datetime widget changes

File: panel/tests/test_links.py [hunk 2]
```diff
@@ -207,11 +207,11 @@ def test_widget_bkplot_link(document, comm):
     assert scatter.glyph.fill_color == '#ff00ff'
     
     code = """
-    value = source['color'];
+    var value = source['color'];
     value = value;
     value = value;
```
Comment (mattpap) [inline panel/tests/test_links.py]: @philippjfr, that's an odd bit of code ^^^.

Comment (philippjfr) [reply panel/tests/test_links.py]: Agreed, it's auto-generated and those empty value assignments are placeholders for possible transforms that may be applied at those stages but are simply no-ops in this case.

Comment (mattpap) [reply panel/tests/test_links.py]: > it's auto-generated

OK, that's the missing bit.

Comment (mattpap) [reply panel/tests/test_links.py]: And it's a test file. I didn't notice that before.

File: setup.py [hunk 3]
```diff
@@ -125,7 +125,7 @@ def run(self):
         'vtk',
         'scikit-learn',
         'datashader',
-        'jupyter_bokeh<=1.1.1',
+        'jupyter_bokeh',
```
Comment (mattpap) [inline setup.py]: I would recommend `>=2.0.0` as it fixes some incompatibilities with bokeh 2.0.

Comment (mattpap) [reply setup.py]: Actually, jupyter_bokeh 2.0 already depends on bokeh 2.0, so scratch that.

**Pull request merged**